Xamarin.Forms.Core/DispatcherExtensions.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace Xamarin.Forms
+{
+	internal static class DispatcherExtensions
+	{
+		static IDispatcherProvider s_current;
+		static IDispatcher s_default;
+
+		public static IDispatcher GetDispatcher(this BindableObject bindableObject)
+		{
+			if (s_default != null)
+			{
+				// If we're already using the fallback dispatcher, keep using it
+				return s_default;
+			}
+
+			// See if the current platform has a DispatcherProvider for us
+			s_current = s_current ?? DependencyService.Get<IDispatcherProvider>();
+
+			if (s_current == null)
+			{
+				// No DispatcherProvider available, use the fallback dispatcher
+				s_default = new FallbackDispatcher();
+				return s_default;
+			}
+
+			// Use the DispatcherProvider to retrieve an appropriate dispatcher for this BindableObject
+			return s_current.GetDispatcher(bindableObject);
+		}
```

Comment:
instead of caching the provider and the default, I would only cache the current Provider...
```csharp
GetDispatcher()
{
    if (s_current == null) {
        var provider = DS.Get<IDP>();
        s_current = provider.?GetDispatcher(bo) ?? new FallBack();
    }
    return s_current;
}